source/base/parameter_handler.cc

```diff
@@ -2968,83 +2967,63 @@ ParameterHandler::scan_line (std::string         line,
             {
               const unsigned int pattern_index
                 = entries->get<unsigned int> (path + path_separator + "pattern");
-              if (!patterns[pattern_index]->match(entry_value))
-                {
-                  std::cerr << "Line <" << current_line_n
-                            << "> of file <" << input_filename
-                            << ">:" << std::endl
-                            << "    The entry value" << std::endl
-                            << "        " << entry_value << std::endl
-                            << "    for the entry named" << std::endl
-                            << "        " << entry_name << std::endl
-                            << "    does not match the given pattern" << std::endl
-                            << "        " << patterns[pattern_index]->description()
-                            << std::endl;
-                  return false;
-                }
+              AssertThrow (patterns[pattern_index]->match (entry_value),
+                           ExcInvalidEntryForPattern (current_line_n,
+                                                      input_filename,
+                                                      entry_value,
+                                                      entry_name,
+                                                      patterns[pattern_index]->description()));
             }
 
           entries->put (path + path_separator + "value",
                         entry_value);
-          return true;
         }
       else
         {
-          std::cerr << "Line <" << current_line_n
-                    << "> of file <" << input_filename
-                    << ">: No such entry was declared:" << std::endl
-                    << "    " << entry_name << std::endl
-                    << "    <Present subsection:" << std::endl;
-          for (unsigned int i=0; i<subsection_path.size(); ++i)
-            std::cerr << std::setw(i*2+8) << " "
-                      << "subsection " << subsection_path[i] << std::endl;
-          std::cerr << "    >" << std::endl;
-
-          return false;
+          Assert(false,
```

Comment:
Assert or AssertThrow (I think the latter)? If you do an Assert, you should do a ``return;`` afterwards.